### PR TITLE
fix: update app id in contract

### PR DESCRIPTION
**Pull request by MatejVukosav**
Fix for https://github.com/calimero-network/core/issues/916

### CODE REVIEW

File: crates/meroctl/src/cli/context/create.rs [hunk 1]
```diff
@@ -260,7 +260,9 @@ async fn update_context_application(
         &format!("admin-api/dev/contexts/{context_id}/application"),
     )?;
 
-    let request = UpdateContextApplicationRequest::new(application_id);
+    //How to make conversion from libp2p to public key
+    let public_key = keypair.public();
+    let request = UpdateContextApplicationRequest::new(application_id, public_key);
```
Comment (miraclx) [inline crates/meroctl/src/cli/context/create.rs]: no, it shouldn't be the libp2p keypair, this should be the `member_public_key` from the create context response

this was wrong about `context call` as well, which I [fixed some days ago](https://github.com/calimero-network/core/pull/948/files#diff-f5d45fa7e6c0662f1afc369ed781cd633d630781f205ce957203e02437621722L77-R97)

File: crates/context/src/lib.rs [hunk 2]
```diff
@@ -849,6 +850,43 @@ impl ContextManager {
             bail!("Context not found")
         };
 
+        let this = self.clone();
```
Comment (miraclx) [inline crates/context/src/lib.rs]: why clone `self`?

File: crates/context/src/lib.rs [hunk 3]
```diff
@@ -834,10 +834,11 @@ impl ContextManager {
...
     }
 
-    pub fn update_application_id(
+    pub async fn update_application_id(
         &self,
         context_id: ContextId,
         application_id: ApplicationId,
+        signer_id: PublicKey,
     ) -> EyreResult<()> {
         // todo! use context config
```
Comment (miraclx) [inline crates/context/src/lib.rs]: we can remove this now

File: crates/context/src/lib.rs [hunk 4]
```diff
@@ -849,6 +850,43 @@ impl ContextManager {
...
+        let Some(application) = this.get_application(&application_id)? else {
+            bail!("Application with id {:?} not found", application_id)
+        };
+
+        let Some(ContextIdentityValue {
+            private_key: Some(requester_secret),
+        }) = handle.get(&ContextIdentityKey::new(context_id, signer_id))?
+        else {
+            bail!(
+                "Identity {:?} not found in context {:?}",
```
Comment (miraclx) [inline crates/context/src/lib.rs]:
```suggestion
                "'{}' is not a member of '{}'",
```

File: crates/context/src/lib.rs [hunk 5]
```diff
@@ -849,6 +850,43 @@ impl ContextManager {
...
+                context_id
+            )
+        };
+
+        let _ = this
+            .config_client
+            .mutate::<ContextConfigEnv>(
+                this.client_config.new.protocol.as_str().into(),
+                this.client_config.new.network.as_str().into(),
+                this.client_config.new.contract_id.as_str().into(),
```
Comment (miraclx) [inline crates/context/src/lib.rs]: we should read the context config from the `ContextConfig` store record

for example

https://github.com/calimero-network/core/blob/db644002b9093b0363e2af55347ebb849161e07e/crates/context/src/lib.rs#L397-L399

File: crates/context/src/lib.rs [hunk 6]
```diff
@@ -849,6 +850,43 @@ impl ContextManager {
...
+            )
+            .update_application(
+                context_id.rt().expect("bla"),
```
Comment (miraclx) [inline crates/context/src/lib.rs]:
```suggestion
                context_id.rt().expect("infallible conversion"),
```

File: crates/context/src/lib.rs (hunk 5 above)
Comment (MatejVukosav) [reply crates/context/src/lib.rs]: updated

**Review by miraclx — APPROVED** [review]

**Pull request merged**